docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,629 @@
+
+
+### 7. Set up remote tracking
+
+The local copy of your repository needs to track both your remote repository and the original repository from which it was forked. By default, when cloning, a link is created back to your copy but not the original repository.
+
+Let's find out what remotes currently exist.
+
+Type
+
+```
+git remote
+```
+
+![submit_pr_remote_tracking_1.png](./imgs/submit_pr_remote_tracking_1.png)
+
+There should be one remote named **origin** and it points to the forked remote repository.
+
+If a remote named **upstream** (pointing to the original remote repository) does not exist, you need to create it.
+
+In your browser, navigate to original repository  <https://github.com/strongloop/loopback-next> and ensure that the `master` branch is selected.
+
+
+![submit_pr_remote_tracking_2.png](./imgs/submit_pr_remote_tracking_2.png)
+
+Click on the `Clone or download` button, leave the cloning option as `Clone with SSH` and press the `copy to clipboard` icon.
+
+![submit_pr_remote_tracking_3.png](./imgs/submit_pr_remote_tracking_3.png)
+
+
+Return to your terminal window and type :
+
+```
+git remote add upstream git@github.com:strongloop/loopback-next.git
+```
+
+then type :
+
+```
+git remote
+```
+
+there should now be two remotes : **origin** and **upstream**.
+
+![submit_pr_remote_tracking_4.png](./imgs/submit_pr_remote_tracking_4.png)
+
+Typing
+
+```
+git remote -v
+```
+
+provides more details.
+
+![submit_pr_remote_tracking_5.png](./imgs/submit_pr_remote_tracking_5.png)
+
+### 8. Install dependencies
+
+The `loopback-next` has several package dependencies.
+
+Type `npm install` to install them.
+
+![submit_pr_npm_install_1.png](./imgs/submit_pr_npm_install_1.png)
+
+
+### 9. Do some work in your local feature branch directory
+
+Whether you are contributing to code or documentation, be sure to make all your changes inside in the local feature branch directory.
+
+Be sure to read, understand, and abide by the instructions provided in [Contributing code](https://loopback.io/doc/en/contrib/code-contrib.html), [Contributing to docs](https://loopback.io/doc/en/contrib/doc-contrib.html), and
+[Contributing to LoopBack](https://github.com/strongloop/loopback-next/blob/master/docs/CONTRIBUTING.md)
+
+### 10. Verifying your code changes
+
+Before `staging your changes` (the next step), be sure your files are free of formatting, syntax, and logical/execution errors.
+
+To flush out any **formatting** issues in code, run `npm run lint`. If there are issues, run `npm run lint:fix`.
+
+To flush out any **syntax** issues in code, run `npm run build`. If there are issues, fix them, and run `npm run build` again. Repeat until all issues are gone.
+
+To flush out any **logical/execution** issues in code, run `npm run test`. This will run the existing mocha test cases, and any new test cases you have added. If there are issues, fix them, run `npm build`, and run `npm run test` again. Repeat until all issues are gone.
```

Comment:
Maybe you could also mention that running `npm test` will run `npm run lint` after if all the tests pass. You could move this one to before `To flush out any **formatting** issues..`? Just a suggestion.

Reply:
Maybe the simplest approach is `npm run lint:fix && npm test`.

Reply:
+1 ^

I usually run `npm test` to make sure all tests pass first then fix the lint error. And `npm run lint` automatically runs as the post test script. 
If tests are all green but `npm run lint` fails, I run `npm run lint:fix` or manually fix the formatting issues(some cannot be fixed by script)

Reply:
And my understanding for command `npm run build` is it mainly generates the codes in `dist` folder.

If you read the scripts in `package.json`: https://github.com/strongloop/loopback-next/blob/master/package.json#L48-L55

`clean` and `build` scripts are run as pretest and `lint` script is run as posttest, which means `npm run test` starts from cleaning your workspace, rebuilds your project to make sure the compiled codes are based on the latest `src` file, runs all tests, finally checks the format.